### PR TITLE
Ks/selectregion

### PR DESCRIPTION
**Pull request by kswanson33**
Add signal holding a view of the rectangular selection to image context & several small bug fixes

### CODE REVIEW

File: src/Tinker.jl [hunk 1]
```diff
@@ -12,21 +25,11 @@ mutable struct ImageContext{T}
     pandrag::Signal{Bool}
     zoomclick::Signal{Bool}
     rectselect::Signal{Bool}
+    shape::Signal{Rectangle} # Holds selection outline
```
Comment (timholy) [inline src/Tinker.jl]: Should this hold any kind of `Shape`?

File: src/Tinker.jl [hunk 2]
```diff
@@ -4,6 +4,19 @@ using Gtk.ShortNames, GtkReactive, Graphics, Colors, Images, IntervalSets
 
 img_ctxs = Signal([])
 
+abstract type Shape end
```
Comment (timholy) [inline src/Tinker.jl:7]: :+1:

**Review by timholy — COMMENTED** [review]
LGTM!

File: src/Tinker.jl (hunk 1 above)
Comment (kswanson33) [reply src/Tinker.jl]: It should! However, it doesn't seem to like it when I say so. I had a similar problem with typing the rectview field (should be the next one after shape). I think it ought to be a signal of type AbstractArray, but again it throws an error when I say so. Do you think the fact that they're signals is causing the problem somehow?

Comment (timholy) [reply src/Tinker.jl]: Maybe try `shape::Signal{<:Shape}`?

Comment (kswanson33) [reply src/Tinker.jl]: Thanks! It works.

File: src/Tinker.jl [hunk 3]
```diff
@@ -12,21 +25,11 @@ mutable struct ImageContext{T}
     pandrag::Signal{Bool}
     zoomclick::Signal{Bool}
     rectselect::Signal{Bool}
+    shape::Signal{<:Shape} # Holds selection outline
+    rectview::Signal{<:AbstractArray} # holds rectangular region corresponding to outline (type?)
```
Comment (timholy) [inline src/Tinker.jl:29]: Glad it worked. Do keep in mind that this Signal will only update when you replace the *container* (the array itself), not the objects inside the container.

**Pull request merged**